pkg/apis/helm.fluxcd.io/v1/types_helmrelease.go

```diff
@@ -407,81 +408,134 @@ type HelmReleaseSpec struct {
+	HelmReleaseReleased HelmReleaseConditionType = "Released"
+	// RolledBack means the chart to which the HelmRelease refers
+	// has been rolled back.
+	HelmReleaseRolledBack HelmReleaseConditionType = "RolledBack"
+)
 
 type HelmReleaseCondition struct {
 	// Type of the condition, one of ('ChartFetched', 'Released', 'RolledBack').
 	Type HelmReleaseConditionType `json:"type"`
+
 	// Status of the condition, one of ('True', 'False', 'Unknown').
 	Status ConditionStatus `json:"status"`
+
 	// LastUpdateTime is the timestamp corresponding to the last status
 	// update of this condition.
 	// +optional
 	LastUpdateTime *metav1.Time `json:"lastUpdateTime,omitempty"`
+
 	// LastTransitionTime is the timestamp corresponding to the last status
 	// change of this condition.
 	// +optional
 	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+
 	// Reason is a brief machine readable explanation for the condition's last
 	// transition.
 	// +optional
 	Reason string `json:"reason,omitempty"`
+
 	// Message is a human readable description of the details of the last
 	// transition, complementing reason.
 	// +optional
 	Message string `json:"message,omitempty"`
 }
 
-// HelmReleaseConditionType represents an HelmRelease condition value.
-// Valid HelmReleaseConditionType values are:
+// HelmReleasePhase represents the phase a HelmRelease is in.
+// Valid HelmReleasePhase values are:
 // "ChartFetched",
-// "Released",
-// "RolledBack"
-// +kubebuilder:validation:Enum="ChartFetched";"Released";"RolledBack"
+// "ChartFetchFailed",
+// "Installing",
+// "Upgrading",
+// "Succeeded",
+// "Failed",
+// "RollingBack",
+// "RolledBack",
+// "RollbackFailed",
+// +kubebuilder:validation:Enum="ChartFetched";"ChartFetchFailed";"Installing";"Upgrading";"Succeeded";"Failed";"RollingBack";"RolledBack";"RollbackFailed"
 // +optional
-type HelmReleaseConditionType string
+type HelmReleasePhase string
 
 const (
 	// ChartFetched means the chart to which the HelmRelease refers
 	// has been fetched successfully
-	HelmReleaseChartFetched HelmReleaseConditionType = "ChartFetched"
-	// Released means the chart release, as specified in this
-	// HelmRelease, has been processed by Helm.
-	HelmReleaseReleased HelmReleaseConditionType = "Released"
-	// RolledBack means the chart to which the HelmRelease refers
-	// has been rolled back
-	HelmReleaseRolledBack HelmReleaseConditionType = "RolledBack"
+	HelmReleasePhaseChartFetched HelmReleasePhase = "ChartFetched"
+	// ChartFetchedFailed means the chart to which the HelmRelease
+	// refers could not be fetched.
+	HelmReleasePhaseChartFetchFailed HelmReleasePhase = "ChartFetchFailed"
+
+	// Installing means the installation for the HelmRelease is running.
+	HelmReleasePhaseInstalling HelmReleasePhase = "Installing"
+	// Upgrading means the upgrade for the HelmRelease is running.
+	HelmReleasePhaseUpgrading HelmReleasePhase = "Upgrading"
+	// Succeeded means the installation or upgrade of the HelmRelease
```

Comment:
Nitpicking:
```suggestion
	// Succeeded means the installation or upgrade for the HelmRelease
```
As the HelmRelease itself is not something you install or upgrade, but rather the Helm release it manages. Same kind of confusion below.